packages/back-end/src/services/auth/index.ts

```diff
@@ -101,15 +104,17 @@ export async function processJWT(
       return true;
     }
 
-    // Generate full list of permissions for the user
-    const userPermissions = getUserPermissions(
-      req.userId,
-      req.organization,
-      teams
-    );
-
-    // Check if the user has the permission
-    return hasPermission(userPermissions, permission, project, envs);
+    if (READ_ONLY_PERMISSIONS.includes(permission)) {
+      // Read only type permissions grant permission if the user has the permission globally or in atleast 1 project
+      return projects.some((p) =>
+        hasPermission(userPermissions, permission, p, envs)
+      );
+    } else {
+      // All other permissions require the user to have the permission globally or the user must have the permission in every project they have specific permissions for
+      return projects.every((p) =>
+        hasPermission(userPermissions, permission, p, envs)
+      );
+    }
```

Comment:
This is the meat of the change. We introduced the concept of `READ_ONLY_PERMISSIONS` - currently those are: readData, runQueries, and viewEvents. When evaluating whether or not the user has permission to do these events, they only have to have the permission in one of the projects they have specific permissions for (or they need to have it globally).

Whereas the remaining permissions require the user to have the permission either globally, or for EVERY project they have project specific permissions for

Reply:
Currently working through 1 issue - for user's who have a global role of `Experimenter`, but have a project-specific role for atleast 1 project of `No Access` the permissions check is failing for `runQueries` when the resource is in `ALL PROJECTS`, as when the `projects` prop is an empty array, we're looping over the user's projects, and adding that to the array. 

And in this case, the permissions for that project is `No Access`.